community/command-line/src/main/java/org/neo4j/commandline/admin/AdminCommand.java

```diff
@@ -60,9 +62,18 @@ public String name()
         }
 
         /**
-         * @return A help string for the command's arguments, if any.
+         * @return The arguments this command accepts.
          */
-        public abstract Optional<String> arguments();
+        public abstract Arguments allArguments();
+
+        /**
+         *
+         * @return A list of possibly mutually-exclusive argument sets for this command.
+         */
+        public List<Arguments> possibleArguments()
+        {
+            return Arrays.asList( allArguments() );
+        }
```

Comment:
Do you think it's practical to combine these two methods into a single method with a more flexible return type? I'm not sure. Maybe anyway this is good enough for now. I just think it's going to be hard for implementors to understand whether they should override `possibleArguments()`. Seems a shame to have all this complexity for that one case, doesn't it?